### PR TITLE
TUI: use vte_version for libvte

### PR DESCRIPTION
**Pull request by blueyed**
Support was added in https://github.com/GNOME/vte/commit/efaf8f3c.

Fixes https://github.com/neovim/neovim/issues/9083.

### CODE REVIEW

File: src/nvim/tui/tui.c [hunk 1]
```diff
@@ -1852,8 +1852,9 @@ static void augment_terminfo(TUIData *data, const char *term,
...
       ut, "ext.disable_mouse", "\x1b[?1002l\x1b[?1006l");
 
-  int ext_bool_Su = unibi_find_ext_bool(ut, "Su");
-  if (ext_bool_Su != -1 && unibi_get_ext_bool(ut, (size_t)ext_bool_Su)) {
+  int ext_bool_Su = unibi_find_ext_bool(ut, "Su");  // used by kitty
+  if (vte_version >= 5102
+      || (ext_bool_Su != -1 && unibi_get_ext_bool(ut, (size_t)ext_bool_Su))) {
       data->unibi_ext.enter_undercurl_mode = (int)unibi_add_ext_str(
           ut, "ext.enter_undercurl_mode", "\x1b[4:3m");
       data->unibi_ext.exit_undercurl_mode = (int)unibi_add_ext_str(
```
Comment (justinmk) [inline src/nvim/tui/tui.c]: the string name (param below this line) doesn't match

Comment (blueyed) [reply src/nvim/tui/tui.c]: Fixed.
It is not used though anywhere anyway.  Should we exit it explicitly somewhere?

**Comment by fmoralesc** [comment]
I can confirm that this works in gnome-terminal 3.30.1 ;) Thanks a lot, @blueyed!

File: src/nvim/tui/tui.c (hunk 1 above)
Comment (justinmk) [reply src/nvim/tui/tui.c]: It is used for debugging, e.g. `nvim -V3log`

**Pull request merged**